pkg/controller/daemon/controller.go

```diff
@@ -671,7 +671,7 @@ func (dsc *DaemonSetsController) syncDaemonSet(key string) error {
 		return err
 	}
 	dsNeedsSync := dsc.expectations.SatisfiedExpectations(dsKey)
-	if dsNeedsSync {
+	if dsNeedsSync && ds.DeletionTimestamp == nil {
```

Comment:
It's where you want to set the pod's Controller ref. I mean you need to stop setting controller ref if the controller is about to be deleted. In daemon set's case, we might do it in addPod().

I'll push my changes to RC later today so I can get your comments, though it's not completed :)